include/seqan3/core/algorithm/detail/algorithm_executor_blocking.hpp

```diff
@@ -313,6 +313,7 @@ class algorithm_executor_blocking
     {
         algorithm = std::move(other.algorithm);
         buffer_size = std::move(other.buffer_size);
+        std::swap(exec_handler, other.exec_handler);
```

Comment:
```suggestion
        using std::swap;
        swap(exec_handler, other.exec_handler);
```

Mhmm, in general the current default move implementation of the parallel execution handler is wrong. 
This should be fixable by adding a custom deleter to the unique ptr, which does what the destructor of the class does: Closing the queue and waiting on the threads to finish. Then the destructor doesn't need to do this.
And this didn't work: `exec_handler = std::move(other.exec_handler)`?

Reply:
Nope, you could do a member initialization here: https://github.com/seqan/seqan3/blob/d37ba83a333992a551ec835097e564d8b9a00604/include/seqan3/core/algorithm/detail/algorithm_executor_blocking.hpp#L124
( `: exec_handler{other.exec_handler}`).

Putting `exec_handler = std::move(other.exec_handler)` at any point doesn't work because the `exec_handler` was already default constructed; and apparently then the destructor doesn't work properly.

Reply:
Ok, I just wanted to make sure. A possible reason for this is that during the move assignment the old resource with the default initiliased thread pool is simply deleted (as if calling `lhs.reset(rhs.release())`). That has side effects since threads that are still joinable are simply destructed. Hence, the unique ptr needs a special deleter that takes care of this properly. The only weird thing is that swap wouldn't do anything different here, since it is done on the wrapper class which does not offer a swap member or swap overload. That's why I am confused that the proposed code should work.